syntax = "proto3";
package bitcoinlike_proto;

message AccountConfiguration {
    bool is_segwit = 1;
    string currency = 2;
}

message CreateAccountRequest {
    string name = 1;
    string xpub = 2;
    uint32 index = 3;
    AccountConfiguration config = 4;
}

message CreateAccountResponse {
    string acc_uid = 1;
}

message SyncAccountRequest {
    string acc_uid = 1;
}

message SyncAccountResponse {
    bool new_acc = 1;
}

message GetBalanceRequest {
    string acc_uid = 1;
}

message GetBalanceResponse {
    string balance = 1;
}

enum BitcoinLikeRequestType {
    CREATE_ACCOUNT = 0;
    SYNC = 1;
    GET_BALANCE = 2;
}

message BitcoinLikeRequest {
    BitcoinLikeRequestType type = 1;
    bytes submessage = 2;
}